kilkari/src/main/java/org/motechproject/nms/kilkari/service/impl/SubscriptionServiceImpl.java

```diff
@@ -690,4 +694,23 @@ public void toggleActiveSubscriptionCapping(MotechEvent event) {
     public List<Subscription> retrieveAll() {
         return subscriptionDataService.retrieveAll();
     }
+
+
+    @Override
+    public void deactivateSubscriptionForSpecificMsisdn(long callingNumber) {
```

Comment:
Not sure why this was chose to exist in subscription service. The phone number is a very subscriber level concept and you want to pull up a subscriber attached to a phone number and then deactivate subscriptions attached to that subscriber.

Ideally, you would deactivate a subscriber attached to a phone number and the result of that being that all the subscriptions attached to the phone number get deactivated.